src/ir/effects.h

```diff
@@ -872,6 +872,18 @@ class EffectAnalyzer {
       if (curr->ref->type.isNullable()) {
         parent.implicitTrap = true;
       }
+      switch (curr->order) {
+        case MemoryOrder::Unordered:
+          break;
+        case MemoryOrder::SeqCst:
+          // Synchronizes with other threads.
+          parent.isAtomic = true;
+          break;
+        case MemoryOrder::AcqRel:
+          // Only synchronizes if other threads can read the field.
+          parent.isAtomic = curr->ref->type.getHeapType().isShared();
```

Comment:
What does SeqCst + unshared mean? That is, what is the meaning of synchronizing with other threads if the other threads cannot read it?

Reply:
The extra power of seqcst accesses is that they form a total global order among themselves no matter what memory they access (or don't, in the case of seqcst fences). This global ordering can be useful on its own:

```
// T1
x = 1
u.store(99, seqcst)
r1 = y

// T2
y = 1
v.store(42, seqcst)
r2 = x
```

In this program it is not possible for `r1` and `r2` to both be 0 because either the store to `u` happens before the store to `v`, in which case `r2` must be 1, or the store to `v1` happens before the store to `u`, in which case `r1` must be 1. Note that it does not matter what addresses `u` and `v` are or whether they are observable by other threads. This example also works if the seqcst stores are just fences or any other seqcst operation. (Credit to https://sabrinajewson.org/rust-nomicon/atomics/fences.html#seqcst-fences for the example)

Reply:
>  Note that it does not matter what addresses u and v are or whether they are observable by other threads.

Hmm, but in the example both T1 and T2 observe both x and y. I am having a hard time imagining a situation where T1 observes only x and T2 only y, but it matters that they are ordered.

(But this question is beyond the scope of this PR I suppose!)

Reply:
Yes, `x` and `y` are observed by both threads, so they must be located in a shared memory  or heap object, but `u` and `v` can just as well live in an unshared memory or heap object.

Reply:
Oh, right, now I see, thanks.